File: Assignment_#1/src/app/src/main/java/com/cpm/g1/theacmeelectronicsshop/dataClasses/basket/Basket.kt
package com.cpm.g1.theacmeelectronicsshop.dataClasses.basket

import java.io.Serializable

data class Basket(
    var userUUID: String = "",
    var products: List<ItemQuantity>,
    var total: String? = "",
    var date: String? = "",
    var hour: String? = "",
): Serializable {
}

File: Assignment_#1/server/api/models/basket.js
const mongoose = require('mongoose');
const uuid = require('node-uuid');

const Basket = mongoose.model('Basket', mongoose.Schema({
    userUUID: String, 
    token: { type: String, default: ()=>{return uuid.v1()}},
    usedToken: {Boolean, default: false},
    products: [{
        id: Number,
        quantity: Number 
    }],
    total: String, 
    date: String,
    hour: String
}));

exports.Basket = Basket; 